flink-optimizer/src/main/java/org/apache/flink/optimizer/plan/Channel.java

```diff
@@ -134,6 +134,10 @@ public Channel(PlanNode sourceNode, TempMode tempMode) {
 	public PlanNode getSource() {
 		return this.source;
 	}
+
+	public void setSource(PlanNode source) {
```

Comment:
Create new channels means that we need to update the input channel existing `PlanNode`, there is no interface to set input channel in `PlanNode` as well.